garglk/cheapglk/cgdate.cpp

```diff
@@ -176,32 +179,47 @@ static glsi32 gli_simplify_time(time_t timestamp, glui32 factor)
 
 void glk_current_time(glktimeval_t *time)
 {
+#ifndef WIN32
     struct timeval tv;
-
     if (gettimeofday(&tv, NULL)) {
         gli_timestamp_to_time(0, 0, time);
         gli_strict_warning("current_time: gettimeofday() failed.");
         return;
     }
-
     gli_timestamp_to_time(tv.tv_sec, tv.tv_usec, time);
+#else
+    struct timespec ts;
+    if (!timespec_get(&ts, TIME_UTC)) {
```

Comment:
`TIME_UTC` isn't defined on MinGW.  I'm guessing the proper approach is to check for MSVC (or for _not_ MinGW), and only do this for that platform.  MinGW should continue to use the POSIX interface.